tests/components/lovelace/test_init.py

```diff
@@ -1,17 +1,169 @@
+        image: https://www.home-assistant.io/images/default-social.png
+        entity: light.bed_light
+
+    # Specify a tab icon if you want the view tab to be an icon.
+  - icon: mdi:home-assistant
+    # Title of the view. Will be used as the tooltip for tab icon
+    title: Second view
+    cards:
+        # Entities card will take a list of entities and show their state.
+      - type: entities
+        # Title of the entities card
+        title: Example
+        # The entities here will be shown in the same order as specified.
+        # Each entry is an entity ID or a map with extra options.
+        entities:
+          - light.kitchen
+          - switch.ac
+          - entity: light.living_room
+            # Override the name to use
+            name: LR Lights
+
+        # The markdown card will render markdown text.
+      - type: markdown
+        title: Lovelace
+        content: >
+          Welcome to your **Lovelace UI**.
+"""
+
+TEST_YAML_B = """\
+title: Home
+views:
+  - title: Dashboard
+    icon: mdi:home
+    cards:
+      - id: testid
+        type: vertical-stack
+        cards:
+          - type: picture-entity
+            entity: group.sample
+            name: Sample
+            image:  /local/images/sample.jpg
+            tap_action: toggle
+"""
+
+# Test data that can not be loaded as YAML
+TEST_BAD_YAML = """\
+title: Home
+views:
+  - title: Dashboard
+    icon: mdi:home
+    cards:
+      - id: testid
+          type: vertical-stack
+"""
+
+
+class TestYAML(unittest.TestCase):
+    """Test lovelace.yaml save and load."""
+
+    def setUp(self):
+        """Set up for tests."""
+        self.tmp_dir = mkdtemp()
+        self.yaml = YAML(typ='rt')
+
+    def tearDown(self):
+        """Clean up after tests."""
+        for fname in os.listdir(self.tmp_dir):
+            os.remove(os.path.join(self.tmp_dir, fname))
+        os.rmdir(self.tmp_dir)
+
+    def _path_for(self, leaf_name):
+        return os.path.join(self.tmp_dir, leaf_name+".yaml")
+
+    def test_save_and_load(self):
+        """Test saving and loading back."""
+        fname = self._path_for("test1")
+        save_yaml(fname, self.yaml.load(TEST_YAML_A))
+        data = load_yaml(fname)
+        self.assertEqual(data, self.yaml.load(TEST_YAML_A))
+
+    def test_overwrite_and_reload(self):
+        """Test that we can overwrite an existing file and read back."""
+        fname = self._path_for("test3")
+        save_yaml(fname, self.yaml.load(TEST_YAML_A))
+        save_yaml(fname, self.yaml.load(TEST_YAML_B))
+        data = load_yaml(fname)
+        self.assertEqual(data, self.yaml.load(TEST_YAML_B))
+
+    def test_load_bad_data(self):
+        """Test error from trying to load unserialisable data."""
+        fname = self._path_for("test5")
+        with open(fname, "w") as fh:
+            fh.write(TEST_BAD_YAML)
+        with self.assertRaises(HomeAssistantError):
+            load_yaml(fname)
+
+    def test_add_id(self):
```

Comment:
These two tests should follow the testing pattern below, as they don't need to do I/O. Doing I/O in tests is bad as it is slow.